### PR TITLE
Fix notes

### PR DESCRIPTION
**Pull request by DrunkenToast**
Make it work in DMs, weird get inconsistenties etc

### CODE REVIEW

**Review by xunafay — CHANGES_REQUESTED** [review]
move notes logic in `command.ts` into seperate file

**Comment by DrunkenToast** [comment]
> move notes logic in `command.ts` into seperate file

Done, ready for review.

File: src/services/notes-service.ts [hunk 1]
```diff
@@ -31,15 +43,100 @@ export class NotesService {
...
+      }
+    }
+    
+    //!notes remove <channel> <number>
+    if(guildConfig) {
+      if (!(msg.member?.hasPermission('ADMINISTRATOR'))) {
+        return 'You have to be an admin to delete notes.';
+      }
+
+      console.log('start');
```
Comment (PandaGamersHDNL) [inline src/services/notes-service.ts]: Rm log

File: src/services/notes-service.ts [hunk 2]
```diff
@@ -31,15 +43,100 @@ export class NotesService {
...
+      if (tokenizer.tokens[1]?.type === 'text' && tokenizer.tokens[1].content === 'remove'
+        && tokenizer.tokens[2]?.type === 'channel' && tokenizer.tokens[3]?.type === 'text') {
+        const noteNum: number = parseInt(tokenizer.tokens[3].content);
+        return this.delChannelNote(noteNum, tokenizer.tokens[2].content.substr(2, 18), guildConfig.id)
+          .catch((err) => {Logger.error(err); return;});
+      }
+      console.log('here');
```
Comment (PandaGamersHDNL) [inline src/services/notes-service.ts]: Rm log

File: src/services/notes-service.ts [hunk 3]
```diff
@@ -48,48 +145,124 @@ export class NotesService {
...
+    }
+    
     // When a server has notes
-    if (guildNotes.notes[channelID] === undefined) {
-      guildNotes.notes[channelID] = []; // create empty array if channel doesnt have notes yet
+    if (notes.notes[channelID] == null) {
+      notes.notes[channelID] = []; // create empty array if channel doesnt have notes yet
+    }
+    notes.notes[channelID].push(note);
+    console.log(notes);
```
Comment (PandaGamersHDNL) [inline src/services/notes-service.ts]: Rm log

**Review by PandaGamersHDNL — CHANGES_REQUESTED** [review]

**Review by xunafay — APPROVED** [review]

**Review by PandaGamersHDNL — APPROVED** [review]

**Pull request merged**